### ISSUE TITLE
fxLayoutGap (and fxFlexOrder) issues

### ISSUE THREAD

**Issue by alexmurari** [ngbracket/ngx-layout#28]
# Bug Report

First of all, thank you very much for continuing the FlexLayout project.
I'm learning Angular (and flex layout) and may be making mistakes here, but I do _believe_ that this might be a bug.

#### What is the expected behaviour?

- `fxLayoutGap` should deactivate correctly.
- `fxLayoutGap` should be aware of `fxFlexOrder` activation to apply margin to the correct elements (in new order).

#### What is the current behaviour?

- `fxLayoutGap` is not deactivating correctly, it's leaving the margin on the element
**Before activating** 
![image](https://github.com/ngbracket/ngx-layout/assets/11204378/36daa0c2-5d04-4a56-8b7f-88030850ba7a)
**After deactivating**
![image](https://github.com/ngbracket/ngx-layout/assets/11204378/e8c0d49f-ea53-4373-9176-e472d984084b)

- `fxLayoutGap` is applying margin to wrong element when using `fxFlexOrder` (the is no margin between the two buttons)
![image](https://github.com/ngbracket/ngx-layout/assets/11204378/deaf7041-0bdd-4e93-a3ff-ed2b3dc6bd39)

#### What are the steps to reproduce?
```html
<div fxLayout="row" fxLayoutAlign="space-evenly center" fxLayout.xs="column" fxLayoutAlign.xs="stretch" fxLayoutGap.xs="20px">
      <button mat-stroked-button fxFlexOrder.xs="2">Forgot Password</button>
      <button mat-raised-button color="primary" type="submit" fxFlexOrder.xs="1">Login</button>
</div>
```
Angular 16.2
Angular Material 16.2.1
ngx-layout 16.0.0

**Comment by DuncanFaulkner** [comment]
Thanks for raising the issue

**Comment by alexmurari** [comment]
@DuncanFaulkner I was able to add failing tests and even fix the layout gap not deactivating correctly.
The other bug about `fxLayoutGap` +  `fxFlexOrder` is giving me a hard time here.

The problem is that the function that retrieves the order to apply the gap is running before the function that applies the `order` style to the element. The scenario where it happens is in the failing test.

Do you have any tips?

Here's the failing test:

```typescript
it('should add gap styles in proper order when order style is applied on responsive layout change', () => {
      let template = `
        <div fxLayout="row" fxLayoutAlign="space-evenly center" fxLayout.xs="column" fxLayoutGap.xs="20px">
          <div fxFlex fxFlexOrder.xs="3"></div>
          <div fxFlex fxFlexOrder.xs="2"></div>
          <div fxFlex fxFlexOrder.xs="1"></div>
        </div>
      `;
      createTestComponent(template);
      mediaController.activate('md');
      fixture.detectChanges();

      let nodes = queryFor(fixture, '[fxFlex]');
      expect(nodes.length).toEqual(3);
      expectEl(nodes[2]).not.toHaveStyle({ 'margin-bottom': '*' }, styler);
      expectEl(nodes[1]).not.toHaveStyle({ 'margin-bottom': '*' }, styler);
      expectEl(nodes[0]).not.toHaveStyle({ 'margin-bottom': '*' }, styler);

      mediaController.activate('xs');
      fixture.detectChanges();
      expectEl(nodes[2]).toHaveStyle({ 'margin-bottom': '20px' }, styler); // fails
      expectEl(nodes[1]).toHaveStyle({ 'margin-bottom': '20px' }, styler);
      expectEl(nodes[0]).not.toHaveStyle({ 'margin-bottom': '20px' }, styler); // fails
      expectEl(nodes[0]).not.toHaveStyle({ 'margin-bottom': '0px' }, styler);
    });
```

**Comment by DuncanFaulkner** [comment]
unfortunately, I don't have any tips at the moment, I will need to look into the issue.

**Comment by alexmurari** [comment]
@DuncanFaulkner I've managed to find a (simple) [solution](https://github.com/ngbracket/ngx-layout/pull/29). Added tests and everything's working fine.

**Comment by DuncanFaulkner** [comment]
this is excellent, I will merge this into the main branch and I'll then update the 16.xx branch

**Comment by DuncanFaulkner** [comment]
merged into main

**Comment by alexmurari** [comment]
@DuncanFaulkner The CI linter wasn't happy that I forgot a whitespace 😆 

Fix: #31

**Comment by alexmurari** [comment]
@DuncanFaulkner Glad to see the build works now. When you have the time, please add the latest version to npm.

**Comment by DuncanFaulkner** [comment]
Yes I also need to push main into 16 branch as well hope I can get it done tonight
